backend/src/backend/db/models.py

```diff
@@ -0,0 +1,129 @@
+    relationship,
+)
+from sqlalchemy.sql.schema import MetaData
+
+from backend.enums import StatusEnum
+
+
+class Base(MappedAsDataclass, DeclarativeBase):
+    # This map details the specific transformation of types between Python and
+    # PostgreSQL. This is only needed for the case where a specific PostgreSQL
+    # type has to be used.
+
+    type_annotation_map = {  # noqa: RUF012
+        str: CITEXT,  # transform Python str to PostgreSQL CITEXT
+        List[str]: ARRAY(
+            item_type=CITEXT
+        ),  # transform Python List[str] into PostgreSQL Array of strings
+        datetime: DateTime(
+            timezone=False
+        ),  # transform Python datetime into PostgreSQL Datetime without timezone
+        IPv4Address: INET,  # transform Python IPV4Address into PostgreSQL INET
+    }
+
+    # This metadata specifies some naming conventions that will be used by
+    # alembic to generate constraints names (indexes, unique constraints, ...)
+    metadata = MetaData(
+        naming_convention={
+            "ix": "ix_%(column_0_label)s",
+            "uq": "uq_%(table_name)s_%(column_0_name)s",
+            "ck": "ck_%(table_name)s_%(constraint_name)s",
+            "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+            "pk": "pk_%(table_name)s",
+        }
+    )
+    pass
+
+
+class Country(Base):
+    __tablename__ = "country"
+
+    code: Mapped[str] = mapped_column(primary_key=True)
+    name: Mapped[str]
+
+    worker_id: Mapped[Optional[UUID]] = mapped_column(
+        ForeignKey("worker.id"), init=False
+    )
+    worker: Mapped[Optional["Worker"]] = relationship(
+        back_populates="countries", init=False
+    )
+    mirrors: Mapped[List["Mirror"]] = relationship(
+        back_populates="country",
+        init=False,
+        cascade="all, delete-orphan",
+    )
+
+    __table_args__ = (UniqueConstraint("name", "code"),)
+
+
+class Mirror(Base):
+    __tablename__ = "mirror"
+
+    base_url: Mapped[str] = mapped_column(primary_key=True)
+    enabled: Mapped[bool]
+    # metadata of a mirror from MirroBrain (https://mirrorbrain-docs.readthedocs.io/en/latest/mirrors.html#displaying-details-about-a-mirror)
+    id: Mapped[Optional[str]]
+    region: Mapped[Optional[str]]
+    asn: Mapped[Optional[str]]
+    score: Mapped[Optional[int]]
+    latitude: Mapped[Optional[float]]
+    longitude: Mapped[Optional[float]]
+    country_only: Mapped[Optional[bool]]
+    region_only: Mapped[Optional[bool]]
+    as_only: Mapped[Optional[bool]]
+    other_countries: Mapped[Optional[List[str]]]
+
+    country_code: Mapped[str] = mapped_column(
+        ForeignKey("country.code"),
+        init=False,
+    )
+    country: Mapped["Country"] = relationship(back_populates="mirrors", init=False)
+
+
+class Worker(Base):
+    __tablename__ = "worker"
+    id: Mapped[UUID] = mapped_column(
+        init=False, primary_key=True, server_default=text("uuid_generate_v4()")
+    )
+    auth_info: Mapped[str]
```

Comment:
I wrote `auth_info` as it was to be decided. We should be more explicit about what this is. I believe we've settled on authenticating via an RSA keypair. Then we'd need to store the public key in PKCS8 and the fingerprint (not mandatory –can be computed– but it's handy)